core/src/sumeragi/fault.rs

```diff
@@ -1033,6 +973,121 @@ pub fn run_sumeragi_main_loop<F>(
                 },
             };
 
+            if maybe_incoming_message.is_some() {
+                let stolen_message = maybe_incoming_message.take().unwrap();
+
+                let peer_list = state_machine_guard
+                    .current_topology
+                    .sorted_peers()
+                    .iter()
+                    .cloned()
+                    .collect();
+                match stolen_message {
+                    Message::TransactionGossip(tx_gossip) => {
+                        for transaction in tx_gossip.txs {
+                            let tx_maybe = VersionedAcceptedTransaction::from_transaction(
```

Comment:
This one I think is a bit different because I think it's best to handle the error case first. A early return becomes confusing if it's written at the end in my opinion.